dartagnan/src/main/java/com/dat3m/dartagnan/parsers/program/visitors/boogie/PthreadsProcedures.java

```diff
@@ -65,6 +68,33 @@ public static void handlePthreadsFunctions(VisitorBoogie visitor, Call_cmdContex
         }
     }
 
+    private static void pthread_join(VisitorBoogie visitor, Call_cmdContext ctx) {
+
+        // ----- TODO: Test code -----
+        if (!visitor.inlineMode) {
+            visitor.addEvent(EventFactory.newFunctionCall("dummy pthread_join()"));
+            // TODO: Create a proper function call, not just an annotation
+            return;
+        }
+        // ----- TODO: Test code end -----
+        // reg := pthread_join(tid);
+        final Register reg = visitor.getScopedRegister(ctx.call_params().Ident(0).getText());
+        Expression threadToJoinWith = (Expression) ctx.call_params().exprs().expr(0).accept(visitor);
+        //FIXME: The first parameter should be a proper register, but we do not know its constant value here
+        // so we loop it up in a map.
+        threadToJoinWith = visitor.expr2tid.getOrDefault(threadToJoinWith, threadToJoinWith);
```

Comment:
If the expression is not in the map, we return the same expression. I assume this is the case where we do not know its constant value. What are the implications of this?

Reply:
The implications are that the subsequent check for a constant value will fail, and the parser throws an error that it cannot statically determine with which thread to join (i.e. dynamic thread joining is not allowed).